app/src/main/java/com/hover/stax/views/StaxTransactionStatusView.kt

```diff
@@ -0,0 +1,78 @@
+package com.hover.stax.views
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.text.Html
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.FrameLayout
+import com.hover.sdk.transactions.Transaction
+import com.hover.stax.R
+import com.hover.stax.databinding.TransactionStatusLayoutBinding
+import com.hover.stax.transactions.StaxTransaction
+
+open class StaxTransactionStatusView(context: Context, attrs: AttributeSet) : FrameLayout(context, attrs) {
+    private val binding: TransactionStatusLayoutBinding
+    private var isFlatView: Boolean = false
+
+    init {
+        getAttrs(context, attrs)
+        binding = TransactionStatusLayoutBinding.inflate(LayoutInflater.from(context), this, true)
+    }
+
+    private fun getAttrs(context: Context, attrs: AttributeSet) {
+        val a = context.theme.obtainStyledAttributes(attrs, R.styleable.StaxTransactionStatusView, 0, 0)
+        try {
+            isFlatView = a.getBoolean(R.styleable.StaxTransactionStatusView_isFlatView, false)
+        } finally {
+            a.recycle()
+        }
+    }
+
+    private fun fillFromAttrs() {
+        if (isFlatView) binding.notificationCard.makeFlatView()
+    }
+
+    fun updateInfo(transaction: StaxTransaction) {
+        when (transaction.status) {
+            Transaction.SUCCEEDED -> setSuccessView(transaction.isRecorded)
+            Transaction.PENDING -> setPendingView(transaction.isRecorded)
+            Transaction.FAILED -> setFailedView(transaction.isRecorded)
+        }
+        fillFromAttrs()
+    }
+
+    @SuppressLint("ResourceAsColor")
+    private fun setSuccessView(isBounty: Boolean) {
```

Comment:
These functions are not DRY. Refactor so that they are one function with the setting of each parameter passed in.

Reply:
Thanks, code is now updated. Kindly review